src/common.speech/Transcription/ConversationManager.ts

```diff
@@ -0,0 +1,187 @@
+
+        try {
+
+            Contracts.throwIfNullOrUndefined(args, "args");
+
+            const languageCode: string = args.getProperty(PropertyId.SpeechServiceConnection_RecoLanguage, ConversationTranslatorConfig.defaultLanguageCode);
+            const nickname: string = args.getProperty(PropertyId.ConversationTranslator_Name);
+            const endpointHost: string = args.getProperty(PropertyId.ConversationTranslator_Host, ConversationTranslatorConfig.host);
+            const correlationId: string = args.getProperty(PropertyId.ConversationTranslator_CorrelationId);
+            const subscriptionKey: string = args.getProperty(PropertyId.SpeechServiceConnection_Key);
+            const subscriptionRegion: string = args.getProperty(PropertyId.SpeechServiceConnection_Region);
+            const authToken: string = args.getProperty(PropertyId.SpeechServiceAuthorization_Token);
+
+            Contracts.throwIfNullOrWhitespace(languageCode, "languageCode");
+            Contracts.throwIfNullOrWhitespace(nickname, "nickname");
+            Contracts.throwIfNullOrWhitespace(endpointHost, "endpointHost");
+
+            const queryParams: IStringDictionary<string> = {};
+            queryParams[ConversationTranslatorConfig.params.apiVersion] = ConversationTranslatorConfig.apiVersion;
+            queryParams[ConversationTranslatorConfig.params.languageCode] = languageCode;
+            queryParams[ConversationTranslatorConfig.params.nickname] = nickname;
+
+            const headers: IStringDictionary<string> = {};
+            if (correlationId) {
+                headers[ConversationTranslatorConfig.params.correlationId] = correlationId;
+            }
+            headers[ConversationTranslatorConfig.params.clientAppId] = ConversationTranslatorConfig.clientAppId;
+
+            if (conversationCode !== undefined) {
+                queryParams[ConversationTranslatorConfig.params.roomId] = conversationCode;
+            } else {
+                Contracts.throwIfNullOrUndefined(subscriptionRegion, ConversationTranslatorConfig.strings.authInvalidSubscriptionRegion);
+                headers[ConversationTranslatorConfig.params.subscriptionRegion] = subscriptionRegion;
+                if (subscriptionKey) {
+                    headers[ConversationTranslatorConfig.params.subscriptionKey] = subscriptionKey;
+                } else if (authToken) {
+                    headers[ConversationTranslatorConfig.params.authorization] = `Bearer ${authToken}`;
+                } else {
+                    Contracts.throwIfNullOrUndefined(subscriptionKey, ConversationTranslatorConfig.strings.authInvalidSubscriptionKey);
+                }
+            }
+
+            const config: IRequestOptions = {};
+            config.headers = headers;
+
+            const endpoint: string = `https://${endpointHost}${ConversationTranslatorConfig.restPath}`;
+
+            // TODO: support a proxy
```

Comment:
>// TODO: support a proxy [](start = 12, length = 24)

Also cert validation.

Reply:
Added a comment to include the cert check too in the future.